plutus-core/plutus-core/src/PlutusCore/Bitwise.hs

```diff
@@ -590,12 +590,20 @@ writeBits bs changelist = case unsafeDupablePerformIO . try $ go of
           pokeByteOff ptr flipIx toWrite
 
 -- | Byte replication, as per [CIP-122](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0122)
-replicateByte :: Int -> Word8 -> BuiltinResult ByteString
+-- We want to cautious about the allocation of huge amounts of memory so we
+-- impose the same length limit that's used in integerToByteString.
+replicateByte :: Integer -> Word8 -> BuiltinResult ByteString
 replicateByte len w8
   | len < 0 = do
       emit "replicateByte: negative length requested"
       evaluationFailure
-  | otherwise = pure . BS.replicate len $ w8
+  | len > integerToByteStringMaximumOutputLength = do
+      emit . pack $ "replicateByte: requested length is too long (maximum is "
+               ++ show integerToByteStringMaximumOutputLength
+               ++ " bytes)"
+      emit $ "Length requested: " <> (pack . show $ len)
+      evaluationFailure
+  | otherwise = pure . BS.replicate (fromIntegral len) $ w8
```

Comment:
> Why the change from Int to Integer and this fromIntegral call?

The reason for that is that we already had the `NumBytesCostedAsNumWords` wrapper for one of the arguments to `integerToByteString` and I needed it again here, but `integerToByteString` takes an `Integer` argument so that's what it was wrapping.   Here the underlying function took an `Int` so I would have needed a separate wrapper (I think).  Maybe we should introduce a version of `fromIntegral` that fails if the argument doesn't fit into an `Int` just to avoid any unpleasant surprises.

Reply:
I've put this back to the way it was before and changed the contents of `NumBytesCostedAsNumWords` to `Int`.